src/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -93,7 +93,7 @@ public ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities)
             // potentially run a diff between the old and new tokens.
             capabilities.SemanticTokensOptions = new SemanticTokensOptions
             {
-                Full = false,
+                Full = true,
```

Comment:
We need some kind of mechanism to allow different clients to either enable this or not.  We do not want to enable this in VSCode (which is what will happen with this PR).

Potentially setting this capability dynamically in response to an option change could work.  Not 100% sure yet.

Reply:
Hm right, that's true. Could we perhaps check if the client capabilities include semantic token ranges, and if not, enable this? Since we get those straight away

Reply:
That definitely sounds like a reasonable approach!  We would always only use range, unless the client does not support range.  Works for me